packages/cockpit/src/transactions/shared/export.js

```diff
@@ -0,0 +1,309 @@
+import {
+  __,
+  always,
+  applySpec,
+  complement,
+  cond,
+  defaultTo,
+  either,
+  head,
+  ifElse,
+  is,
+  isEmpty,
+  isNil,
+  join,
+  juxt,
+  last,
+  length,
+  map,
+  path,
+  pathOr,
+  pathSatisfies,
+  pipe,
+  pick,
+  prop,
+  props,
+  propEq,
+  propIs,
+  propOr,
+  propSatisfies,
+  splitAt,
+  T,
+  unless,
+  values,
+} from 'ramda'
+import moment from 'moment'
+
+const LIMITER = '-'
+
+// eslint-disable-next-line no-useless-escape
+const scapeString = value => `"${value}"`
+
+const isArrayToStringScape = value => ifElse(
+  is(Array),
+  pipe(
+    map(propOr('', value)),
+    join(', '),
+    scapeString
+  ),
+  always(LIMITER)
+)
+
+const isEmptyOrNill = either(isNil, isEmpty)
+
+const propOrLimiter = propOr(LIMITER)
+
+const isAntifraudScoreNil = propSatisfies(isNil, 'antifraud_score')
+
+const isRefuseReasonNil = propSatisfies(isNil, 'refuse_reason')
+
+const getCardProp = subProp => cond([
+  [
+    pathSatisfies(complement(isNil), ['card', subProp]),
+    path(['card', subProp]),
+  ],
+  [
+    propSatisfies(complement(isNil), `card_${subProp}`),
+    prop(`card_${subProp}`),
+  ],
+  [T, always(LIMITER)],
+])
+
+const isRefuseReasonAntifraud = propEq('refuse_reason', 'antifraud')
+
+const antifraudRecommendation = cond([
+  [isAntifraudScoreNil, always(LIMITER)],
+  [isRefuseReasonNil, always(LIMITER)],
+  [isRefuseReasonAntifraud, always('refused')],
+  [T, always('approved')],
+])
+
+const getAntifraudProp = ifElse(
+  isAntifraudScoreNil,
+  always(LIMITER),
+  applySpec({
+    recommendation: antifraudRecommendation,
+    score: propOrLimiter('antifraud_score'),
+  })
+)
+
+const pickCardDigits = pick(['card_first_digits', 'card_last_digits'])
+
+const concatCardDigits = pipe(
+  pickCardDigits,
+  values,
+  join('******')
+)
+
+const getCardNumber = ifElse(
+  propIs(String, 'card_first_digits'),
+  concatCardDigits,
+  always(LIMITER)
+)
+
+const getCustomerName = pipe(path(['customer', 'name']))
+
+const getCustomerSubProp = subProp => pathOr(LIMITER, ['customer', subProp])
+
+const getAddressSubProp = subProp => pathOr(LIMITER, [path(['billing', 'address']), subProp])
```

Comment:
Fazendo alguns testes por aqui notei que os endereços não estão sendo coletados por esta função, consegui corrigir quebrando ela em duas:

```suggestion
const getAddressProp = either(
  path(['billing', 'address']),
  path(['address'])
)

const getAddressSubProp = property => 
  either(
    pipe(
      getAddressProp, 
      prop(property)
    ), 
    () => LIMITER
  )
```